pkg/providers/imagefamily/nodeimage.go

```diff
@@ -38,10 +38,10 @@ type NodeImageProvider interface {
 
 // Returns the list of available NodeImages for the given AKSNodeClass sorted in priority ordering
 func (p *Provider) List(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass) ([]NodeImage, error) {
-	kubernetesVersion, err := p.KubeServerVersion(ctx)
-	if err != nil {
+	if err := nodeClass.ValidateKubernetesVersionReadiness(); err != nil {
```

Comment:
minor: You do this "check then read" pattern a few places. Maybe just make a helper method for it on `nodeClass`?

`func (c *nodeClass) GetKubernetesVersion() (string, error)` that calls `ValidateKubernetesVersionReadiness` internally to produce the error, and then returns the version?

Reply:
Seems good. My patterning/idea was that we would always be checking for validity when we read.

Reply:
You could even not export the validate one, and just use the `_, err` variant in that case (avoids 2 methods). IMO either is fine.

Reply:
Updated, and kept the validate one internal.